src/Driver/OCI8/Exception/Error.php

```diff
@@ -16,12 +18,26 @@
  */
 final class Error extends AbstractException
 {
+    private const CODE_TRANSACTION_ROLLED_BACK = 2091;
+
     /** @param resource $resource */
     public static function new($resource): self
     {
         $error = oci_error($resource);
         assert($error !== false);
 
-        return new self($error['message'], null, $error['code']);
+        $code    = $error['code'];
+        $message = $error['message'];
+        if ($code === self::CODE_TRANSACTION_ROLLED_BACK) {
```

Comment:
I don't feel comfortable with making this exception class aware of specific error codes. That logic belongs into `ExceptionConverter`. Can we replace this logic with a generic detection of chained errors?